### PR TITLE
Improve alingment

### PR DESCRIPTION
**Pull request by akaDuality**
Can align by center, by several axes and the line become red!

### CODE REVIEW

File: Document/Sources/Document/Drawing/Alignment/AlignmentDirection.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import CoreGraphics
+
+public enum AlignmentDirection: CaseIterable {
```
Comment (drucelweisse) [inline Document/Sources/Document/Drawing/Alignment/AlignmentDirection.swift:3]: Looks like a [UnitPoint](https://developer.apple.com/documentation/swiftui/unitpoint). Maybe we can benefit to if we get inspiration of that API

File: Document/Sources/Document/Drawing/Alignment/Overlay/AlignmentOverlay.swift [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    
+    func alignToAny(
+        _ sourceControl: A11yControl,
+        point: CGPoint,
+        drawnControls: [A11yControl]
+    ) -> CGPoint {
+        
+        hideAligningLine()
+        
+        let alignments: [AlignmentPoint] = drawnControls
```
Comment (drucelweisse) [inline Document/Sources/Document/Drawing/Alignment/Overlay/AlignmentOverlay.swift:57]: is it the same as line 84? maybe we can extract filtering to some function

**Review by drucelweisse — APPROVED** [review]

File: Document/Sources/Document/Drawing/Alignment/AlignmentDirection.swift (hunk 1 above)
Comment (akaDuality) [reply Document/Sources/Document/Drawing/Alignment/AlignmentDirection.swift:3]: Looks interesting. I'll investigate later

File: Document/Sources/Document/Drawing/Alignment/Overlay/AlignmentOverlay.swift (hunk 2 above)
Comment (akaDuality) [reply Document/Sources/Document/Drawing/Alignment/Overlay/AlignmentOverlay.swift:57]: Both function can be union but I'm afraid of difficult abstraction

**Pull request merged**